app/jinja_filters.py

```diff
@@ -93,22 +99,29 @@ def format_unit_input_label(unit: str, unit_length: UnitLengthType = "short") ->
     :param (str) unit_length length of unit text, can be one of short/long/narrow
     """
     unit_label: str
+    # mass-metric-ton no longer supported for en_GB and related locales, but still present in business schema and allowed in validator,
+    # until removed from schema we substitute mass-ton for mass-metric-ton and vice versa, before/after unit formatting
+    measurement_unit = "mass-ton" if unit == "mass-metric-ton" else unit
+
     if unit_length == "long":
         unit_label = units.format_unit(
```

Comment:
Could this be replaced with 
```
unit_label = format_unit(
  value=2,
  unit=unit,
  length=unit_length
).replace("2, ", "")
```
and likewise for the label below so that the hack only has to be in one place?

Reply:
I am not sure, that "2" there is only to get some extra whitespace for long unit. How to make it only call format unit once?

Reply:
so I think you could do this
```
unit_label: str

    if unit_length == "long":
        unit_label = format_unit(value=2, unit=unit, length=unit_length).replace(
            "2 ", ""
        )
    else:
        # Type ignore: We pass an empty string  as the value so that we just return the unit label
        unit_label = format_unit(
            value="", unit=unit, length=unit_length  # type: ignore
        ).strip()

    return unit_label
```

Reply:
and then `format_unit` will have already made the substitution

Reply:
both functions were doing `locale = flask_babel.get_locale()` so the refactor shouldn't affect that

Reply:
Ok, makes perfect sense, sorry, was hard to digest the first time.

Reply:
np! I realised after saying it looked almost the same so I should have added that extra info as a note 😅